src/server/replica.cc

```diff
@@ -354,80 +360,58 @@ error_code Replica::ConnectAndAuth(std::chrono::milliseconds connect_timeout_ms)
+      RETURN_BAD_MESSAGE("Failed authentication with masters");
   }
   return error_code{};
 }
 
 error_code Replica::Greet() {
   parser_.reset(new RedisParser{false});
-  base::IoBuf io_buf{128};
   ReqSerializer serializer{sock_.get()};
-  uint32_t consumed = 0;
   VLOG(1) << "greeting message handling";
   // Corresponds to server.repl_state == REPL_STATE_CONNECTING state in redis
   RETURN_ON_ERR(SendCommand("PING", &serializer));  // optional.
-  RETURN_ON_ERR(ReadRespReply(&io_buf, &consumed));
-  if (!CheckRespIsSimpleReply("PONG")) {
-    LOG(ERROR) << "Bad pong response " << ToSV(io_buf.InputBuffer());
-    return make_error_code(errc::bad_message);
-  }
-
-  io_buf.ConsumeInput(consumed);
+  RETURN_ON_ERR(ReadRespReply());
+  if (!CheckRespIsSimpleReply("PONG"))
+    RETURN_BAD_MESSAGE("Bad pong response");
 
   // Corresponds to server.repl_state == REPL_STATE_SEND_HANDSHAKE condition in replication.c
   auto port = absl::GetFlag(FLAGS_port);
   RETURN_ON_ERR(SendCommand(StrCat("REPLCONF listening-port ", port), &serializer));
-  RETURN_ON_ERR(ReadRespReply(&io_buf, &consumed));
-  if (!CheckRespIsSimpleReply("OK")) {
-    LOG(ERROR) << "Bad REPLCONF response " << ToSV(io_buf.InputBuffer());
-    return make_error_code(errc::bad_message);
-  }
-
-  io_buf.ConsumeInput(consumed);
+  RETURN_ON_ERR(ReadRespReply());
+  if (!CheckRespIsSimpleReply("OK"))
+    RETURN_BAD_MESSAGE("Bad REPLCONF response");
 
   // Corresponds to server.repl_state == REPL_STATE_SEND_CAPA
   RETURN_ON_ERR(SendCommand("REPLCONF capa eof capa psync2", &serializer));
-  RETURN_ON_ERR(ReadRespReply(&io_buf, &consumed));
-  if (!CheckRespIsSimpleReply("OK")) {
-    LOG(ERROR) << "Bad REPLCONF response " << ToSV(io_buf.InputBuffer());
-    return make_error_code(errc::bad_message);
-  }
-
-  io_buf.ConsumeInput(consumed);
+  RETURN_ON_ERR(ReadRespReply());
+  if (!CheckRespIsSimpleReply("OK"))
+    RETURN_BAD_MESSAGE("Bad REPLCONF response");
 
   // Announce that we are the dragonfly client.
   // Note that we currently do not support dragonfly->redis replication.
   RETURN_ON_ERR(SendCommand("REPLCONF capa dragonfly", &serializer));
-  RETURN_ON_ERR(ReadRespReply(&io_buf, &consumed));
-  auto err_handler = [&io_buf]() {
-    LOG(ERROR) << "Unexpected response " << ToSV(io_buf.InputBuffer());
-    return make_error_code(errc::bad_message);
-  };
+  RETURN_ON_ERR(ReadRespReply());
 
   if (!CheckRespFirstTypes({RespExpr::STRING}))
-    return err_handler();
+    RETURN_BAD_MESSAGE("Bad response");
 
   string_view cmd = ToSV(resp_args_[0].GetBuf());
 
   if (resp_args_.size() == 1) {  // Redis
     if (cmd != "OK")
-      return err_handler();
+      RETURN_BAD_MESSAGE("Bad response");
   } else if (resp_args_.size() >= 3) {  // it's dragonfly master.
     if (auto ec = HandleCapaDflyResp(); ec)
-      return err_handler();
+      RETURN_BAD_MESSAGE("Bad response");
     if (auto ec = ConfigureDflyMaster(); ec)
       return ec;
```

Comment:
why we dont use the macro here?

Reply:
we already print the error message inside